src/gui/sdl_mapper.cpp

```diff
@@ -2858,17 +2860,72 @@ static bool load_binds_from_file(const std::string_view mapperfile_path,
 
 void MAPPER_CheckEvent(SDL_Event *event)
 {
-	for (auto &group : bindgroups)
-		if (group->CheckEvent(event))
+	switch (event->type) {
+	case SDL_CONTROLLERDEVICEADDED:
+	case SDL_CONTROLLERDEVICEREMOVED:
+	case SDL_JOYDEVICEREMOVED:
+	case SDL_JOYDEVICEADDED:
+		if (get_joystick_section()->Get_bool("hotplug")) {
+			MAPPER_HandleJoyDeviceEvent(
+			        reinterpret_cast<SDL_JoyDeviceEvent*>(event));
+		}
+		return;
+	default: break;
+	}
+
+	for (auto& group : bindgroups) {
+		if (group->CheckEvent(event)) {
 			return;
+		}
+	}
 }
 
-void BIND_MappingEvents() {
+void MAPPER_HandleJoyDeviceEvent(SDL_JoyDeviceEvent* event)
+{
+	SDL_Joystick* joystick;
+	switch (event->type) {
+	case SDL_CONTROLLERDEVICEADDED:
+	case SDL_JOYDEVICEADDED: {
+		auto index = event->which;
+		joystick   = SDL_JoystickOpen(index);
+		auto name  = SDL_JoystickName(joystick);
+		LOG_INFO("MAPPER: Game controller '%s' has been plugged in", name);
+		break;
+	}
+	case SDL_CONTROLLERDEVICEREMOVED:
+	case SDL_JOYDEVICEREMOVED: {
+		joystick  = SDL_JoystickFromInstanceID(event->which);
+		auto name = SDL_JoystickName(joystick);
+		LOG_INFO("MAPPER: Game controller '%s' has been disconnected", name);
+		break;
+	}
+	default:
+		LOG_ERR("MAPPER: FATAL: MAPPER_HandleJoyDeviceEvent: Unexpected event type '%d' received",
```

Comment:
This is a bit of a mix between a debug message and an error message. User-facing errors shouldn't be programmer-centic, so mentioning SDL internals is a no go. However, I can see how this would be useful as a debug-level message.

So you could definitely could turn this into a debug level message (and get rid of "FATAL: ", please); the other question is do we need to tell the user anything in this case? Probably not.

Reply:
Made it debug, got rid of `FATAL: `